build/build.sh

```diff
@@ -39,7 +49,25 @@ if [[ -n "$TRAVIS_TAG" ]] && [[ $TRAVIS_TAG != *"RC"* ]]; then
 fi
 
 # Get the version details
-VERSION="$(cat $GOPATH/src/github.com/openebs/cstor-operators/VERSION)"
+#VERSION="$(cat $GOPATH/src/github.com/openebs/cstor-operators/VERSION)"
+
+## Populate the version based on release tag
+## If travis tag is set then assign it as VERSION and
+## if travis tag is empty then mark version as ci
+if [ -n "$TRAVIS_TAG" ]; then
+    # Trim the `v` from the TRAVIS_TAG if it exists
+    # Example: v1.10.0 maps to 1.10.0
+    # Example: 1.10.0 maps to 1.10.0
+    # Example: v1.10.0-custom maps to 1.10.0-custom
+    VERSION="${TRAVIS_TAG#v}"
+else
+    VERSION="dev"
```

Comment:
let's make it VERSION="$CURRENT_BRANCH-$SHORT_COMMIT:$BUILDDATE".

Reply:
- This VERSION is used to represent the version of OpenEBS via labels and
version details(IMO going with commit based doesn't fit here).
- Fetching [version](https://github.com/openebs/cstor-operators/blob/3b13054649044aae928c258ee7c160dea8254bb6/pkg/version/version.go#L107) from file and we are fetching version using `GetVersion` in multiple places.

Reply:
GetVersion is executed runtime , this file should be present in the docker file. We don't need this file.

Reply:
>This VERSION is used to represent the version of OpenEBS via labels and
version details(IMO going with commit based doesn't fit here).
for development, we can have the branch and the commit hash from which we have comiled it code. After this change all the local compiled code will show dev as VERSION.

Reply:
> GetVersion is executed runtime , this file should be present in the docker file. We don't need this file.

We are statically linking during compilation time https://github.com/openebs/cstor-operators/blob/3b13054649044aae928c258ee7c160dea8254bb6/build/build.sh#L133

Reply:
sai that is variable, file was just used to set VERSION variable. The file is not used anywhere else.

Reply:
Correct I also just now realized while discussing with Prateek.

Reply:
>After this change all the local compiled code will show dev as VERSION.

Correct!! So how about having branch_name-dev i.e if the branch is v1.11.x then label will be `1.11.0-dev`

Reply:
The reason why I am not going via commit based is we have few [checks](https://github.com/openebs/cstor-operators/blob/3b13054649044aae928c258ee7c160dea8254bb6/pkg/controllers/cstorvolumeconfig/controller.go#L847) in the code if checks mismatch then reconciliation will not happen.

`CVC.VersionDetails.Status.Current` will be populated by [CSI-Plugin](https://github.com/openebs/cstor-csi/blob/68bb336e9777f139127b75c79b5be55dfee906dd/pkg/utils/maya.go#L110) if it mismatch then CStorVolume resources will not be created.

Reply:
did not get that check part entirly, but lets us go ahead with branch_name-dev.

Reply:
Basically that check was added to stop dataplane upgrade when control plane alone is upgraded. I will go ahead with branch_name-dev

Reply:
Raised a PR https://github.com/openebs/cstor-operators/pull/106 after PR got merge then I will cherry-pick commit into this PR